### ISSUE TITLE
VLC doesn't work

### ISSUE THREAD

**Issue by ceskyDJ** [ImranR98/Obtainium#758]
It gives me an "No APK found" error, when I try to add VLC to the list of apps watched by Obtainium (see screenshot attached below).

**HW/SW info:**
- **Device:** Samsung Galaxy S22
- **OS:** Android 13
- **App version:** 0.13.23

**Screenshot:**
- ![Screenshot_20230807_232719](https://github.com/ImranR98/Obtainium/assets/48548230/5a5eea7d-6f1d-4d32-8589-094c99cdd405)

**Comment by Ozomei** [comment]
F-Droid repo is working and that is official and listed on VLC's site 
![IMG20230815122410.jpg](https://github.com/ImranR98/Obtainium/assets/135990135/97a15756-1ba3-4489-ae22-f5a9c2f68c0a)

![Screenshot_2023-08-15-12-20-50-98_a13e17678c56528bd2f18fcaaa8af4ba.jpg](https://github.com/ImranR98/Obtainium/assets/135990135/2ea3fdd7-3d31-472b-81d2-b5192fef6cae)

![Screenshot_2023-08-15-12-21-29-30_a13e17678c56528bd2f18fcaaa8af4ba.jpg](https://github.com/ImranR98/Obtainium/assets/135990135/d776ab7b-bbb7-4479-9811-e915ee45f723)

**Comment by ceskyDJ** [comment]
Thanks for your note. This issue is about what doesn't work, not what does work. Obtainium has support for VLC using its official APK mirror (https://get.videolan.org/vlc-android/3.5.4/), see [README](https://github.com/ImranR98/Obtainium/blob/main/README.md) for details.

**Comment by ceskyDJ** [comment]
> Hello, I confirm the issue. I also tried to bypass the specific support for VLC, using the HTML source with the url: https://get.videolan.org/vlc-android/last/ Does not work either
> 
> **App version**: 0.13.24-beta

You're absolutely right! I noticed it, too, but I forgot to write it here. I think it's another problem (maybe unrelated to VLC support), so I'd consider creating a new issue for that (and because you came with it as the first one, it's fully yours).

**Comment by ImranR98** [comment]
> Hello, I confirm the issue. I also tried to bypass the specific support for VLC, using the HTML source with the url: https://get.videolan.org/vlc-android/last/ Does not work either
> 
> **App version**: 0.13.24-beta

Yes, I don't add apps as their own sources if the generic HTML source can handle them.

**Comment by SleepzCat** [comment]
I got an "Unexpected Error" when pasting in the source link for VLC.

It's not letting me post the image so here's a catbox link: https://files.catbox.moe/2s3a2g.png

**Comment by ImranR98** [comment]
https://github.com/ImranR98/Obtainium/releases/tag/v0.13.26-beta

**Comment by ImranR98** [comment]
> It's not letting me post the image so here's a catbox link: https://files.catbox.moe/2s3a2g.png

The URL should be `videolan.org`

**Comment by ceskyDJ** [comment]
Thanks!

**Comment by akovia** [comment]
Sorry that I understand if this was supposed to be fixed, or if you have to use the source. 
I'm getting error.
"Could not determine release version"

**Comment by ceskyDJ** [comment]
> Sorry that I understand if this was supposed to be fixed, or if you have to use the source. I'm getting error. "Could not determine release version"

What configuration do you use for VLC in Obtainium?

**Comment by SleepzCat** [comment]
> > It's not letting me post the image so here's a catbox link: https://files.catbox.moe/2s3a2g.png
> 
> The URL should be `videolan.org`

No, videolan.org hosts an older version of VLC at that web URL. The web URL I was using is where the publish their latest version.

**Comment by akovia** [comment]
@SleepzCat You didn't quote me, so not sure if you were telling me I have an older version or not.

The version it downloaded for me was 3.5.4-arm64-v8a. That seems to be the same when I go directly to their site to download. If that message wasn't for me, I apologize.

**Comment by SleepzCat** [comment]
> @SleepzCat You didn't quote me, so not sure if you were telling me I have an older version or not.
> 
> The version it downloaded for me was 3.5.4-arm64-v8a. That seems to be the same when I go directly to their site to download. If that message wasn't for me, I apologize.

No, I'm telling Imran that videolan.org hosts an outdated version.

But yes, your version is an old version. The current stable version is 3.5.7. The latest version is 3.6.0 Beta 3

**Comment by Sgt-Nukem** [comment]
> I'm currently using f-droid for VLC, since it's the only way to keep VLC up to date automatically. Imran keeps closing issues related to VLC and telling people to use videolan.org, which is outdated.

Thank you for the info. I just had the time to check out why Obtainium always tells me "Cannot install an older version of an App" for VLC. Now that explains it.

Because I'm avoiding F-Droid where I can for reasons explained in [this video](https://www.youtube.com/watch?v=IzpVI4zaso0) I am now directly using the Play Store for that and have it removed from Obtainium.

The whole point of Obtainium is to update your apps directly from the source - not to keep ancient versions with unfixed security issues. I'm a bit disappointed.

**Comment by ImranR98** [comment]
May have been some misunderstanding if I closed an issue that shouldn't have been (let me know when that happens). Looking at the config on the apps website, you're right - it gives 3.6.0 instead of 3.6.3.

Use this config instead (I'll update the site with it later): https://apps.obtainium.imranr.dev/redirect?r=obtainium://app/%7B%22id%22%3A%22org.videolan.vlc%22%2C%22url%22%3A%22https%3A%2F%2Fget.videolan.org%2Fvlc-android%2F%22%2C%22author%22%3A%22get.videolan.org%22%2C%22name%22%3A%22VLC%22%2C%22preferredApkIndex%22%3A0%2C%22additionalSettings%22%3A%22%7B%5C%22intermediateLink%5C%22%3A%5B%7B%5C%22customLinkFilterRegex%5C%22%3A%5C%22%5E%5B0-9%5D%5C%22%2C%5C%22filterByLinkText%5C%22%3Atrue%2C%5C%22skipSort%5C%22%3Afalse%2C%5C%22reverseSort%5C%22%3Afalse%2C%5C%22sortByLastLinkSegment%5C%22%3Afalse%7D%2C%7B%5C%22customLinkFilterRegex%5C%22%3A%5C%22arm64-v8a.apk%24%5C%22%2C%5C%22filterByLinkText%5C%22%3Afalse%2C%5C%22skipSort%5C%22%3Afalse%2C%5C%22reverseSort%5C%22%3Afalse%2C%5C%22sortByLastLinkSegment%5C%22%3Afalse%7D%5D%2C%5C%22customLinkFilterRegex%5C%22%3A%5C%22%5C%22%2C%5C%22filterByLinkText%5C%22%3Afalse%2C%5C%22skipSort%5C%22%3Afalse%2C%5C%22reverseSort%5C%22%3Atrue%2C%5C%22sortByLastLinkSegment%5C%22%3Afalse%2C%5C%22versionExtractWholePage%5C%22%3Afalse%2C%5C%22requestHeader%5C%22%3A%5B%7B%5C%22requestHeader%5C%22%3A%5C%22User-Agent%3A%20Mozilla%2F5.0%20(Linux%3B%20Android%2010%3B%20K)%20AppleWebKit%2F537.36%20(KHTML%2C%20like%20Gecko)%20Chrome%2F114.0.0.0%20Mobile%20Safari%2F537.36%5C%22%7D%5D%2C%5C%22defaultPseudoVersioningMethod%5C%22%3A%5C%22partialAPKHash%5C%22%2C%5C%22trackOnly%5C%22%3Afalse%2C%5C%22versionExtractionRegEx%5C%22%3A%5C%22vlc-android%2F(%5B%5E%2F%5D%2B)%2F%5C%22%2C%5C%22matchGroupToUse%5C%22%3A%5C%221%5C%22%2C%5C%22versionDetection%5C%22%3Atrue%2C%5C%22useVersionCodeAsOSVersion%5C%22%3Afalse%2C%5C%22apkFilterRegEx%5C%22%3A%5C%22%5C%22%2C%5C%22invertAPKFilter%5C%22%3Afalse%2C%5C%22autoApkFilterByArch%5C%22%3Atrue%2C%5C%22appName%5C%22%3A%5C%22%5C%22%2C%5C%22shizukuPretendToBeGooglePlay%5C%22%3Afalse%2C%5C%22allowInsecure%5C%22%3Afalse%2C%5C%22exemptFromBackgroundUpdates%5C%22%3Afalse%2C%5C%22skipUpdateNotifications%5C%22%3Afalse%2C%5C%22about%5C%22%3A%5C%22%5C%22%2C%5C%22refreshBeforeDownload%5C%22%3Atrue%7D%22%2C%22overrideSource%22%3Anull%7D

**Comment by Sgt-Nukem** [comment]
> I have updated the site with a config that should grab the latest version 3.6.3: https://apps.obtainium.imranr.dev/?categories=video_player

That was quick, thanks.
I uninstalled the Play Store version again and re-added it to Obtainium. And it seems to work.

Only difference I saw is the Play Store version gives this:
![Image](https://github.com/user-attachments/assets/ffc76a01-44d7-4a71-83a3-af7e0435fb39)

While the version installed by Obtainium shows:
![Image](https://github.com/user-attachments/assets/5390d82f-2787-4df6-9d72-b6d7fbae1d12)

_**Notice the higher `versionCode` on Play Store.**_
But that could be because of the way VLC does their different releases, right?
